library/core/src/ffi/c_str.rs

```diff
@@ -125,37 +125,33 @@ pub struct CStr {
 /// ```
 #[derive(Clone, PartialEq, Eq, Debug)]
 #[stable(feature = "core_c_str", since = "1.64.0")]
-pub struct FromBytesWithNulError {
-    kind: FromBytesWithNulErrorKind,
-}
-
-#[derive(Clone, PartialEq, Eq, Debug)]
-enum FromBytesWithNulErrorKind {
+pub enum FromBytesWithNulError {
```

Comment:
this makes the precise enum variants part of the public interface, which we don't want. that's why it's  a private fueld

Reply:
Ah, now I understand. Thanks for the explanation! I'll close this PR.